substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -172,6 +172,9 @@ pub mod pallet {
 		/// issuance.
 		type Reward: OnUnbalanced<PositiveImbalanceOf<Self>>;
 
+		/// Origin from which deprecation calls must come.
+		type DeprecationOrigin: EnsureOrigin<Self::RuntimeOrigin>;
```

Comment:
Any reason to add a temp new origin instead of using `T::AdminOrigin`?

Reply:
Only the Root is assigned to AdminOrigin, so using it would fail in the StakingAdmin track. 

If suitable we could just amend AdminOrigin to take `EitherOr<EnsureRoot, StakingAdmin >` instead. It is used for `cancel_deferred_slash` and `set_min_commission` currently. Is it reasonable for these calls to also go through StakingAdmin?